plutus-ir/test/transform/letFloat/rec2.golden

```diff
@@ -0,0 +1,44 @@
+[
+  (lam
+    x
+    (all a (type) (fun a a))
+    [
+      (lam
+        y
+        (all a (type) (fun a a))
+        (let
+          (rec)
+          (termbind
+            (strict) (vardecl j (con integer)) [ [ (builtin addInteger) x ] k ]
+          )
+          (termbind
+            (strict) (vardecl k (con integer)) [ [ (builtin addInteger) y ] j ]
+          )
+          (let
+            (nonrec)
+            (termbind
+              (strict)
+              (vardecl i1 (con integer))
+              [ [ (builtin addInteger) k ] (con 3) ]
+            )
+            (termbind (strict) (vardecl i2 (con integer)) (con 3))
```

Comment:
I don't like this reshuffling. Imaging the user writing something like this (pseudosyntax):

```
let oneHalfOfTheResult = if sanityCheck then someValue else error
let theOtherHalfOfTheResult = doingLotsOfStuff
oneHalfOfTheResult + theOtherHalfOfTheResult
```

Now if we transform this into

```
let theOtherHalfOfTheResult = doingLotsOfStuff
let oneHalfOfTheResult = if sanityCheck then someValue else error
oneHalfOfTheResult + theOtherHalfOfTheResult
```

then the sanity check runs after `doingLotsOfStuff`, which denotationally doesn't matter, but operationally it's lots of wasted work. What's worse is that the user would never expect that, just imaging debugging something like this.

Admittedly, this is a rather contrived use case, but I really don't like messing with semantics like that. If you present a language as strict-with-error, you're not supposed to reorder user-written `let`s.

Should be discussed at least.

Pinging @michaelpj